### PR TITLE
Request access flasher confirmation

### PR DESCRIPTION
**Pull request by defong**
### Context
Confirmation that request access has been successful.

### Changes proposed in this pull request
On the course page a confirmation message is shown to a user after a successful request access has been submitted.

### Guidance to review
The message will only be shown once, ie navigating away or refreshing page message goes away.

image

Validation summary
image

### CODE REVIEW

File: src/ui/Controllers/RequestAccessController.cs [hunk 1]
```diff
@@ -1,48 +1,43 @@
...
 using GovUk.Education.ManageCourses.ApiClient;
 using GovUk.Education.ManageCourses.Ui.ViewModels;
 using Microsoft.AspNetCore.Authorization;
 using Microsoft.AspNetCore.Mvc;
 
-namespace GovUk.Education.ManageCourses.Ui.Controllers
-{
-    public class RequestAccessController : CommonAttributesControllerBase
-    {   
+namespace GovUk.Education.ManageCourses.Ui.Controllers {
```
Comment (timabell) [inline src/ui/Controllers/RequestAccessController.cs]: This file should be reformatted to the usual c# layout.

**Comment by timabell** [comment]
I think we should use https://github.com/lurumad/core-flash that @dankmitchell found

File: src/ui/Styles/patterns/_success-summary.scss [hunk 2]
```diff
@@ -0,0 +1,22 @@
+.error-summary-heading:only-child {
```
Comment (timabell) [inline src/ui/Styles/patterns/_success-summary.scss]: Could you comment where this code came from? Was it hand cranked or pulled from somewhere? Putting it in the commit comment would suffice.

Comment (dankmitchell) [reply src/ui/Styles/patterns/_success-summary.scss]: I'm going to clean this up quickly

File: src/ui/Controllers/RequestAccessController.cs (hunk 1 above)
Comment (defong) [reply src/ui/Controllers/RequestAccessController.cs]: Fixed

**Comment by defong** [comment]
With regards to https://github.com/lurumad/core-flash not sure about that as it uses bootstrap underneath, particular when there is only one use case to pull in a whole external dependency.

File: src/ui/Views/Courses/Shared/RequestAccessSuccess.cshtml [hunk 3]
```diff
@@ -0,0 +1,9 @@
+﻿@if (TempData["RequestAccess_To_Name"] != null) {
```
Comment (timabell) [inline src/ui/Views/Courses/Shared/RequestAccessSuccess.cshtml]: I would like to see a strongly typed wrapper around TempData that represents our usages.
Something like https://stackoverflow.com/a/10956414/10245 perhaps

**Pull request merged**